pkg/network/tracer/tracer_linux_test.go

```diff
@@ -385,6 +385,8 @@ func TestConnectionExpirationRegression(t *testing.T) {
 }
 
 func TestConntrackExpiration(t *testing.T) {
+	t.Skip("flaky test")
```

Comment:
flaky? I don't think I've had problems with this test failing.

Reply:
gah, you're right. I ran the test multiple times against master and indeed it doesn't fail

Reply:
I think I know what the problem is: the test uses holds a reference to a connection that due to the buffering change, becomes invalid after multiple calls to `GetConnections`.
I'm wondering if I should change this helper to return a copy of the value, to prevent folks from getting into this kind of error in the future:
https://github.com/DataDog/datadog-agent/blob/eaf8fe6246936d4cc8410c5b33feee67eaef7a3c/pkg/network/tracer/tracer_test.go#L692-L698

Reply:
Actually, I'm not sure. The value should be already a copy due to how searchConnections is implemented 🤔 
https://github.com/DataDog/datadog-agent/blob/eaf8fe6246936d4cc8410c5b33feee67eaef7a3c/pkg/network/tracer/tracer_test.go#L700-L708

Reply:
Do we need some kind of guarantee that a buffer isn't attempted to be used again until after it has been reclaimed? Otherwise we a race condition in https://github.com/DataDog/datadog-agent/blob/310dc66b73eedc8b4916fd3afa1ac5604ad95504/cmd/system-probe/modules/network_tracer.go#L66 right?

Reply:
I thought about this, but the way I implemented the `clientPool` should prevent this from happening:
https://github.com/DataDog/datadog-agent/blob/pedro/client-buffer/pkg/network/client_pool.go#L26-L30
So if someone misuses the pool and does something like
```
b1 := clientPool.Get("clientID")
b2 := clientPool.Get("clientID")
Reclaim(b1)
Reclaim(b2)
```
it won't share the same buffer

Reply:
OK, the problem is not the clientBuffer, but another optimization: https://github.com/DataDog/datadog-agent/pull/9265/commits/b4bb82aa8517947ea93537782f206298988e2ceb#diff-b3a5827b0c9e4fad1178e080974122e0aca65a7e724645f7c6e8dd80407ee17aR190-R204

The problem is that the filter/callback is called with the same `*ConnectionStats` object, but when there is a expired connection the callback adds that pointer to a slice and later on it uses it to delete translations